### PR TITLE
Issue #112: Add implementation for `add` command in DG

### LINKED ISSUE #112: Update DG on implementation for `add` command
Week 10's task - DG individual draft.

### PR DESCRIPTION
**Pull request by domlimm**
Closes #112.

### CODE REVIEW

**Review by charmainehly — COMMENTED** [review]
LGTM! :) Thanks for the quick addition!!! Perhaps our team can look into standardising our implementation section in the DG over the next week as we update all the implementations :)

**Review by charmainehly — APPROVED** [review]

**Comment by domlimm** [comment]
@charmainehly thanks for reviewing! will be looking to add more stuff later!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -154,6 +154,42 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+### Add feature
+
+The `add` mechanism is facilitated by `AddressBook`. The implementation of adding a `Candidate` through the `add` command has been enhanced on the existing approach. It extends `Command`. The input parameters of the `add` command has been tweaked slightly, introducing a few more related entities.
+
+List of new entities:
+1. `StudentId`
+2. `Course`
+3. `Seniority`
+4. `ApplicationStatus`
+5. `InterviewStatus`
```
Comment (tiewweijian) [inline docs/DeveloperGuide.md:166]: Maybe you can add in your notes that `ApplicationStatus` and `InterviewStatus` will be set to default. And then maybe in the `EditCommand` I could elaborate further how to edit these attributes

**Review by tiewweijian — APPROVED** [review]

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -154,6 +154,42 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Add feature
+
+The `add` mechanism is facilitated by `AddressBook`. The implementation of adding a `Candidate` through the `add` command has been enhanced on the existing approach. It extends `Command`. The input parameters of the `add` command has been tweaked slightly, introducing a few more related entities.
```
Comment (charmainehly) [inline docs/DeveloperGuide.md:159]: @domlimm I created a PR with the documentation for `find` command! I included a rough layout I thought of to help segment up the different parts to address e.g. _what is the implementation_, _how we implemented it_, _why we decided to implement it that way_ and _relevant UML diagrams_. I'm not sure if it'll be helpful for your part as well - I can update mine accordingly to follow the eventual standardised format! :)

Comment (charmainehly) [reply docs/DeveloperGuide.md:159]: I noted on your comment that you'll be updating the PR further - so sorry missed that out earlier! I'll re-look subsequently (if needed) before merging! :sweat_smile:

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (domlimm) [reply docs/DeveloperGuide.md:166]: Will do @tiewweijian!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (domlimm) [reply docs/DeveloperGuide.md:159]: @charmainehly no worries at all! i will take a look at it! thanks char!

**Comment by domlimm** [comment]
@charmainehly @tiewweijian hey yboth, i've changed it according to char's format and wj's suggestion. please take a look! if all good, please proceed and merge. thank you both! 😍

**Comment by domlimm** [comment]
@charmainehly @tiewweijian heyyo fam, we can merge this in now. all should be good. yall can maybe take a quick peek at the files changed once more before doing so, thanks!

**Comment by domlimm** [comment]
@charmainehly hey char, thanks for sharing and looking it up! 😭 i totally understand! you are not understanding it wrongly. i'll follow your workflow in the future, probably ask you even then! 😛 i think i did `git pull` on the current branch i.e. `112-X`. maybe that's the cause. ugh...! anyway thanks again!

btw, feel free to merge it in! 🚀

**Pull request merged**